### PR TITLE
Inventory Event: Device updated + FIX account when update device.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
Consume kafka message for Inventory updated event and fix the device account when updating the device.

Fixes # THEEDGE-1729

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [X] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `go fmt ./...` to check that my code is properly formatted
- [X] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/routes/updates.go [hunk 1]
```diff
@@ -286,12 +286,17 @@ func updateFromHTTP(w http.ResponseWriter, r *http.Request) (*models.UpdateTrans
 				"deviceUUID": device.ID,
 			}).Info("Creating a new device on the database")
 			updateDevice = &models.Device{
-				UUID: device.ID,
+				UUID:    device.ID,
```
Comment (acosferreira) [inline pkg/routes/updates.go:289]: @ldjebran why we are setting the UUID as ID, shouldnt the ID be the db id of device and UUID be the identifier across the platform?

File: pkg/routes/updates.go [hunk 2]
```diff
@@ -286,12 +286,17 @@ func updateFromHTTP(w http.ResponseWriter, r *http.Request) (*models.UpdateTrans
...
+				Account: account,
 			}
 			db.DB.Create(&updateDevice)
```
Comment (thom-at-redhat) [inline pkg/routes/updates.go:292]: Can this throw an error?

File: pkg/services/consumers.go [hunk 3]
```diff
@@ -125,23 +125,32 @@ func (s *KafkaConsumerService) ConsumeInventoryCreateEvents() error {
...
 		}
-		if eventType == "created" {
-			log.WithFields(log.Fields{
-				"topic":  m.Topic,
-				"offset": m.Offset,
-				"key":    string(m.Key),
-				"value":  string(m.Value),
-			}).Debug("Read message from Kafka topic")
+		if eventType != InventoryEventTypeCreated && eventType != InventoryEventTypeUpdated {
+			log.Debug("Skipping message - not a created and not an updated message from platform insights")
```
Comment (thom-at-redhat) [inline pkg/services/consumers.go:132]: how about something like "This Kafka message is not a platform insights created or updated message -- skipping"?

File: pkg/services/devices.go [hunk 4]
```diff
@@ -396,16 +403,56 @@ func (s *DeviceService) GetDeviceLastDeployment(device inventory.Device) *invent
 	return nil
 }
 
+// ProcessPlatformInventoryUpdatedEvent processes messages from platform.inventory.events kafka topic with event_type="updated"
+func (s *DeviceService) ProcessPlatformInventoryUpdatedEvent(message []byte) error {
+	var eventData PlatformInsightsCreateUpdateEventPayload
+	if err := json.Unmarshal(message, &eventData); err != nil {
+		s.log.WithFields(log.Fields{"value": string(message), "error": err}).Debug("Skipping message - it is not an update message")
```
Comment (thom-at-redhat) [inline pkg/services/devices.go:410]: Similar to above comment, please clarify the action of skipping is due to the Kafka message not matching the desired criteria.   Initially, I read it as "Skipping message" and thought "What does this mean?"

File: pkg/services/devices.go [hunk 5]
```diff
@@ -396,16 +403,56 @@ func (s *DeviceService) GetDeviceLastDeployment(device inventory.Device) *invent
...
+		return err
+	}
+	if eventData.Type != InventoryEventTypeUpdated && eventData.Host.SystemProfile.HostType != "edge" {
+		s.log.Debug("Skipping kafka message - not an edge and not update message from platform insights")
```
Comment (thom-at-redhat) [inline pkg/services/devices.go:414]: better

File: pkg/services/devices.go [hunk 6]
```diff
@@ -396,16 +403,56 @@ func (s *DeviceService) GetDeviceLastDeployment(device inventory.Device) *invent
...
-	var e *PlatformInsightsCreateEventPayload
+	var e *PlatformInsightsCreateUpdateEventPayload
 	err := json.Unmarshal(message, &e)
 	if err != nil {
 		log.WithFields(log.Fields{
 			"value": string(message),
 		}).Debug("Skipping message - it is not a create message" + err.Error())
 	} else {
-		if e.Type == "created" && e.Host.SystemProfile.HostType == "edge" {
+		if e.Type == InventoryEventTypeCreated && e.Host.SystemProfile.HostType == "edge" {
```
Comment (thom-at-redhat) [inline pkg/services/devices.go:455]: Do we need to parameterize HostType "edge"?

File: pkg/services/consumers.go (hunk 3 above)
Comment (ldjebran) [reply pkg/services/consumers.go:132]: We need to start by "Skipping" word in the Log, and not as the last word.

**Comment by Avielyo10** [comment]
@Mergifyio rebase

File: pkg/services/devices.go (hunk 4 above)
Comment (ldjebran) [reply pkg/services/devices.go:410]: It means that we process only "updated" message type.

File: pkg/services/devices.go (hunk 6 above)
Comment (ldjebran) [reply pkg/services/devices.go]: resolved

**Comment by ldjebran** [comment]
Many thanks @thom-at-redhat for your review, all commets has been applied.

**Review by thom-at-redhat — APPROVED** [review]
Nice work!

**Pull request merged**